mlflow/sklearn/__init__.py

```diff
@@ -884,12 +1006,77 @@ def patched_fit(original, self, *args, **kwargs):
                           for autologging (e.g., specify "fit" in order to indicate that
                           `sklearn.linear_model.LogisticRegression.fit()` is being patched)
         """
+        status = _get_autolog_training_status()
         with _SklearnTrainingSession(clazz=self.__class__, allow_children=False) as t:
             if t.should_log():
-                return fit_mlflow(original, self, *args, **kwargs)
+                status.register_model(self, mlflow.active_run().info.run_id)
+                with status.call_scope(status.fit_scope_key):
+                    result = fit_mlflow(original, self, *args, **kwargs)
+                return result
             else:
                 return original(self, *args, **kwargs)
 
+    def patched_predict(original, self, *args, **kwargs):
+        if _SklearnTrainingSession.in_top_level():
+            predict_result = original(self, *args, **kwargs)
+            status = _get_autolog_training_status()
+            if status.should_log_eval_metrics() and \
+                    status.get_run_id_for_model(self):
+                eval_dataset = args[0] if len(args) >= 1 else kwargs.get('X')
+                eval_dataset_name = _inspect_dataset_var_name(eval_dataset)
+                status.register_prediction_result(
+                    self, eval_dataset_name, predict_result)
+            return predict_result
+        else:
+            return original(self, *args, **kwargs)
+
+    def patched_metric_api(original, *args, **kwargs):
+        if _SklearnTrainingSession.in_top_level():
```

Comment:
See explanation in https://github.com/mlflow/mlflow/pull/4491/files#r668771365

Reply:
Can we add an inline comment with this explanation?